apps/token/src/routes/home/index.tsx

```diff
@@ -1,147 +1,199 @@
-import React from 'react';
-import { Trans, useTranslation } from 'react-i18next';
+import { useEffect, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
 import { Link } from 'react-router-dom';
-
-import { Heading, SubHeading } from '../../components/heading';
-import { ExternalLinks } from '@vegaprotocol/react-helpers';
-import { useAppState } from '../../contexts/app-state/app-state-context';
+import {
+  AsyncRenderer,
+  Button,
+  ExternalLink,
+  RoundedWrapper,
+} from '@vegaprotocol/ui-toolkit';
 import { useDocumentTitle } from '../../hooks/use-document-title';
-import type { RouteChildProps } from '..';
+import { ProposalsListItem } from '../proposals/components/proposals-list-item';
 import Routes from '../routes';
-import { TokenDetails } from './token-details';
-import { Button } from '@vegaprotocol/ui-toolkit';
-import { toBigNum } from '@vegaprotocol/react-helpers';
-import { useNodeDataQuery } from './__generated___/NodeData';
+import {
+  ExternalLinks,
+  removePaginationWrapper,
+} from '@vegaprotocol/react-helpers';
+import { useNodesQuery } from '../staking/home/__generated___/Nodes';
+import { useProposalsQuery } from '../proposals/proposals/__generated__/Proposals';
+import { getNotRejectedProposals } from '../proposals/proposals/proposals-container';
+import { Heading } from '../../components/heading';
+import * as Schema from '@vegaprotocol/types';
+import type { RouteChildProps } from '..';
+import type { ProposalFieldsFragment } from '../proposals/proposals/__generated__/Proposals';
+
+const nodesToShow = 6;
 
-const Home = ({ name }: RouteChildProps) => {
+const GovernanceHome = ({ name }: RouteChildProps) => {
   useDocumentTitle(name);
   const { t } = useTranslation();
-  const { appState } = useAppState();
-  const { data } = useNodeDataQuery();
-  const totalAssociated = React.useMemo(() => {
-    return toBigNum(data?.nodeData?.stakedTotal || '0', appState.decimals);
-  }, [appState.decimals, data?.nodeData?.stakedTotal]);
+  const {
+    data: proposalsData,
+    loading: proposalsLoading,
+    error: proposalsError,
+  } = useProposalsQuery({
+    pollInterval: 5000,
+    fetchPolicy: 'network-only',
+    errorPolicy: 'ignore',
+  });
+
+  const {
+    data: validatorsData,
+    error: validatorsError,
+    loading: validatorsLoading,
+    refetch,
+  } = useNodesQuery();
+
+  useEffect(() => {
+    const epochInterval = setInterval(() => {
+      if (!validatorsData?.epoch.timestamps.expiry) return;
+      const now = Date.now();
+      const expiry = new Date(validatorsData.epoch.timestamps.expiry).getTime();
+
+      if (now > expiry) {
+        refetch();
+        clearInterval(epochInterval);
+      }
+    }, 10000);
+
+    return () => {
+      clearInterval(epochInterval);
+    };
+  }, [refetch, validatorsData?.epoch.timestamps.expiry]);
```

Comment:
question (blocking): what is the purpose of this subscription?

Reply:
Oh i see this is to refresh the nodes if the epoch rolls over. Can we make this a generalised hook and share it with the epoch-data component rather than copy/paste please :pray: